tubular/numeric.py

```diff
@@ -518,3 +525,190 @@ def transform(self, X):
+            if svd_solver not in ["auto", "full", "arpack", "randomized"]:
+                raise ValueError(
+                    f"{self.classname()}:svd_solver {svd_solver} is unknown. Please select among 'auto', 'full', 'arpack', 'randomized'."
+                )
+            else:
+                self.svd_solver = svd_solver
+        else:
+            raise TypeError(
+                f"{self.classname()}:unexpected type {type(svd_solver)} for svd_solver, must be str"
+            )
+
+        if type(random_state) is int:
+            self.random_state = random_state
+        else:
+            if random_state is None:
+                self.random_state = random_state
+            else:
+                raise TypeError(
+                    f"{self.classname()}:unexpected type {type(random_state)} for random_state, must be int or None."
+                )
+
+        self.pca = PCA(
+            n_components=self.n_components,
+            svd_solver=self.svd_solver,
+            random_state=self.random_state,
+        )
+        self.feature_names_out = None
+        self.n_components_ = None
+
+    def check_numeric_columns(self, X):
+        """Method to check all columns (specicifed in self.columns) in X are all numeric.
+
+        Parameters
+        ----------
+        X : pd.DataFrame
+            Data containing columns to check.
+
+        """
+
+        numeric_column_types = X[self.columns].apply(
+            pd.api.types.is_numeric_dtype, axis=0
+        )
+
+        if not numeric_column_types.all():
+
+            non_numeric_columns = list(
+                numeric_column_types.loc[~numeric_column_types].index
+            )
+
+            raise TypeError(
+                f"{self.classname()}: The following columns are not numeric in X; {non_numeric_columns}"
+            )
+
+        return X
+
+    def fit(self, X, y=None):
+        """Fit PCA to input data.
+
+        Parameters
+        ----------
+        X : pd.DataFrame
+            Dataframe with columns to learn scaling values from.
+
+        y : None
+            Required for pipeline.
+
+        """
+
+        super().fit(X, y)
+
+        X = self.check_numeric_columns(X)
+
+        self.pca.fit(X[self.columns])
+        self.n_components_ = self.pca.n_components_
+        self.feature_names_out = ["pca" + str(i) for i in range(self.n_components_)]
+
+        return self
+
+    def transform(self, X):
+        """Generate from input pandas DataFrame (X) PCA features and add this column or columns in X.
+        Parameters
+        ----------
+        X : pd.DataFrame
+            Data to transform.
+        Returns
+        -------
+        X : pd.DataFrame
+            Input X with additional column or columns (self.interaction_colname) added. These contain the output of
+            running the  product pandas DataFrame method on identified combinations.
+        """
+        X = super().transform(X)
+        X = self.check_numeric_columns(X)
```

Comment:
again - is assignment necessary?

Reply:
related to a new git issue check_numeric_columns ?